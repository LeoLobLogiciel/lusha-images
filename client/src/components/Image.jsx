import { useState } from "react"
import ImageDetail from "./ImageDetails"

const Image = ({image}) => {

  const [hover, setHover] =  useState(false)
  
  return (
    <div>
      <article
        key={image.id}
        onMouseEnter={ () => setHover(true)}
        onMouseLeave={ () => setHover(false)}
      >
        {hover ? <ImageDetail image={image} /> : <img src={image.url} />}
        
      </article>

    </div>
)
}

export default Image