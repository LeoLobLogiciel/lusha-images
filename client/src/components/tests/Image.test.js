import React from 'react';
import Image from '../Image';
import renderer from 'react-test-renderer'
import { shallow, mount, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'


configure({adapter: new Adapter()});

const mockProps = {
  image: {
    url: "testUrl",
    id: 1
  }
};

test("Snapshot test", () => {
  const component = renderer.create(
    <Image image={mockProps.image} />
  )
  let tree=component.toJSON()
  expect(tree).toMatchSnapshot()

  const componentShallow = shallow(<Image image={mockProps.image} />)

})
