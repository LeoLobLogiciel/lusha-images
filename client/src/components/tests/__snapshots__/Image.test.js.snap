// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div>
  <article
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      src="testUrl"
    />
  </article>
</div>
`;
